apps/console/src/features/users/components/add-user-updated.tsx

```diff
@@ -0,0 +1,947 @@
+        "usernameSymbols");
+    const USERNAME_HAS_INVALID_LENGTH_ERROR_MESSAGE: string =
+        t("extensions:manage.features.user.addUser.validation.usernameLength", {
+            maxLength: usernameConfig?.maxLength,
+            minLength: usernameConfig?.minLength
+        });
+
+    const eventPublisher: EventPublisher = EventPublisher.getInstance();
+
+    /**
+     * Set the password setup option to 'create-password'.
+     */
+    useEffect(() => {
+        if (!passwordOption) {
+            if (!requestedPasswordOption) {
+                setPasswordOption("create-password");
+
+                return;
+            }
+
+            setPasswordOption(requestedPasswordOption);
+        }
+    }, [ requestedPasswordOption ]);
+
+    useEffect(() => {
+        if (passwordOption === "create-password") {
+            setUserSummaryEnabled(true);
+            setAskPasswordFromUser(true);
+            setOfflineUser(false);
+        } else {
+            if (askPasswordOption === "offline") {
+                setUserSummaryEnabled(true);
+                setAskPasswordFromUser(false);
+                setOfflineUser(true);
+
+                return;
+            }
+            setUserSummaryEnabled(false);
+            setAskPasswordFromUser(false);
+            setOfflineUser(false);
+        }
+    }, [ passwordOption, askPasswordOption ]);
+
+    useEffect(() => {
+        if (validationConfig) {
+            setPasswordConfig(getConfiguration(validationConfig));
+            setUsernameConfig(getUsernameConfiguration(validationConfig));
+        }
+    }, [ validationConfig ]);
+
+    useEffect(() => {
+        if (!isAlphanumericUsernameEnabled()) {
+            setIsValidEmail(true);
+        } else {
+            setIsValidEmail(false);
+            // setAskPasswordOption("offline");
+        }
+    }, [ usernameConfig ]);
+
+    /**
+     * Check whether the alphanumeric usernames are enabled.
+     *
+     * @returns isAlphanumericUsernameEnabled - validation status.
+     */
+    const isAlphanumericUsernameEnabled= (): boolean => {
+
+        return usernameConfig?.enableValidator === "true";
+    };
+
+    /**
+     * Callback function to validate password.
+     *
+     * @param valid - validation status.
+     * @param validationStatus - detailed validation status.
+     */
+    const onPasswordValidate = (valid: boolean): void => {
+
+        setIsValidPassword(valid);
+    };
+
+    /**
+     * The following function gets the user store regex that validates user name.
+     */
+    const getUserStoreRegex = async () => {
+        setRegExLoading(true);
+        await SharedUserStoreUtils.getUserStoreRegEx(userstore,
+            SharedUserStoreConstants.USERSTORE_REGEX_PROPERTIES.UsernameRegEx)
+            .then((response: string) => {
+                setUserStoreRegex(response);
+                setRegExLoading(false);
+            });
+    };
+
+    useEffect(() => {
```

Comment:
Let's group all useEffect s to one place

Reply:
Fixed with fd5839362baf8e4e50397013a20cfa8b0b15fa4e